### PR TITLE
Refactor wallet store

### PR DESCRIPTION
**Pull request by ashchan**
Remove electron-store and refactor wallet store to separate its main wallet list file and keystore files for each wallet.

## Breaking Notice

It will not load your existing data!!!

### CODE REVIEW

**Comment by Keith-CY** [comment]
CI failed

**Comment by ashchan** [comment]
Resolving conflicts...

File: packages/neuron-wallet/src/services/wallets.ts [hunk 1]
```diff
@@ -1,66 +1,191 @@
...
+    fs.writeFileSync(this.storeLocation(), JSON.stringify(keystore), { encoding: 'utf8' })
+  }
+
+  deleteKeystore = () => {
+    fs.unlinkSync(this.storeLocation())
+  }
+}
+
+// TODO: Check if '/dev/wallets' path works on Windows
+const defaultStorePath = env.isDevMode ? '/dev/wallets' : '/wallets'
```
Comment (classicalliu) [inline packages/neuron-wallet/src/services/wallets.ts]: maybe can use `path`

Comment (Keith-CY) [reply packages/neuron-wallet/src/services/wallets.ts]: `path` module in Node.js

File: packages/neuron-wallet/src/services/wallets.ts [hunk 2]
```diff
@@ -1,66 +1,191 @@
...
 export default class WalletService {
-  public getAll = (): WalletData[] => {
-    return walletStore.getAllWallets()
+  private storePath: string
+  private listStore: Store // Save wallets (meta info except keystore, which is persisted separately)
+  private walletsKey = 'wallets'
+  private currentWalletKey = 'current'
+
+  constructor(storePath: string = defaultStorePath) {
+    this.storePath = `${app.getPath('userData')}/${storePath}`
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/services/wallets.ts]: use `path.join`

File: packages/neuron-wallet/src/services/wallets.ts [hunk 3]
```diff
@@ -1,66 +1,191 @@
...
-      throw Error('Wallet name existed')
+  public get = (id: string): Wallet | undefined => {
+    const wallet = this.getAll().find(w => w.id === id)
+    if (wallet) {
+      return FileKeystoreWallet.fromJSON(wallet, this.storePath)
     }
-    const id = v4()
-    walletStore.saveWallet({ id, name, keystore, addresses })
-    return { id, name, keystore, addresses }
+    return undefined
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/services/wallets.ts]: better to use `null` which explicitly means `none`

File: packages/neuron-wallet/src/services/wallets.ts [hunk 4]
```diff
@@ -1,66 +1,191 @@
...
   }
 
-  public setActive = (id: string): boolean => {
-    return walletStore.setActiveWallet(id)
+  public getCurrent = (): Wallet | undefined => {
+    const walletId = this.listStore.readSync(this.currentWalletKey) as string
+    if (walletId) {
+      return this.get(walletId)
+    }
+    return undefined
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/services/wallets.ts]: better to use `null`

**Comment by Keith-CY** [comment]
conflicts

File: packages/neuron-wallet/src/services/wallets.ts (hunk 3 above)
Comment (ashchan) [reply packages/neuron-wallet/src/services/wallets.ts]: Makes sense. Doing that in a future refactoring.

**Pull request merged**